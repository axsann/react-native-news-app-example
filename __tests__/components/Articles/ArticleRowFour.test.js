// @flow

import 'react-native'
import React from 'react'
import ArticleRowFour from '../../../src/components/Articles/ArticleRowFour.component'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const mockFourItems = [
  {"title":"あああああああああああああああああああああああああああ","url":"https://example.com","date":"20XX年Y月Z日","imgUrl":"https://placehold.jp/150x150.png","tags":["タグB"]},
  {"title":"あああああああああああああああああああああああああああ","url":"https://example.com","date":"20XX年Y月Z日","imgUrl":"https://placehold.jp/150x150.png","tags":["タグA","タグG"]},
  {"title":"あああああああああああああああああああああああああああ","url":"https://example.com","date":"20XX年Y月Z日","imgUrl":"https://placehold.jp/150x150.png","tags":["タグE"]},
  {"title":"あああああああああああああああああああああああああああ","url":"https://example.com","date":"20XX年Y月Z日","imgUrl":"https://placehold.jp/150x150.png","tags":["タグC","タグA"]},
]

describe('ArticleRowFour', () => {
  it('renders correctly', () => {
    const instance = renderer.create(
      <ArticleRowFour articles={mockFourItems} onPressArticle={()=>{}}  />
    )

    expect(instance.toJSON()).toMatchSnapshot()
  })
})
