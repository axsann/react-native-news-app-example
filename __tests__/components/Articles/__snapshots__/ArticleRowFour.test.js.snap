// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleRowFour renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "height": 190,
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://placehold.jp/150x150.png",
            }
          }
          style={
            Object {
              "flex": 8,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "flex": 9,
              },
              Object {
                "marginTop": 10,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
                "fontSize": 12.5,
                "fontWeight": "bold",
              }
            }
          >
            あああああああああああああああああああああああああああ
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 10,
                "marginTop": 5,
              }
            }
          >
            20XX年Y月Z日
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#eeeeee",
          "width": 1,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://placehold.jp/150x150.png",
            }
          }
          style={
            Object {
              "flex": 8,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "flex": 9,
              },
              Object {
                "marginTop": 10,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
                "fontSize": 12.5,
                "fontWeight": "bold",
              }
            }
          >
            あああああああああああああああああああああああああああ
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 10,
                "marginTop": 5,
              }
            }
          >
            20XX年Y月Z日
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#eeeeee",
          "width": 1,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://placehold.jp/150x150.png",
            }
          }
          style={
            Object {
              "flex": 8,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "flex": 9,
              },
              Object {
                "marginTop": 10,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
                "fontSize": 12.5,
                "fontWeight": "bold",
              }
            }
          >
            あああああああああああああああああああああああああああ
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 10,
                "marginTop": 5,
              }
            }
          >
            20XX年Y月Z日
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eeeeee",
          "height": 1,
        },
        Object {
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "height": 100,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontWeight": "bold",
          }
        }
      >
        あああああああああああああああああああああああああああ
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 10,
            "marginTop": 5,
          }
        }
      >
        20XX年Y月Z日
      </Text>
    </View>
    <Image
      source={
        Object {
          "uri": "https://placehold.jp/150x150.png",
        }
      }
      style={
        Object {
          "flex": 2,
        }
      }
    />
  </View>
</View>
`;
